athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchFieldResolver.java

```diff
@@ -0,0 +1,275 @@
+
+    protected ElasticsearchFieldResolver() {}
+
+    /**
+     * Return the field value from a complex structure or list.
+     * @param field is the field that we would like to extract from the provided value.
+     * @param originalValue is the original value object.
+     * @return the field's value as an ArrayList for a LIST field type, a HashMap for a STRUCT field type, or the actual
+     * value if neither of the above.
+     * @throws RuntimeException if the fieldName does not exist in originalValue, if the fieldType is a STRUCT and
+     * the fieldValue is not instance of HashMap, or if the fieldType is neither a LIST or a STRUCT but the fieldValue
+     * is instance of HashMap (STRUCT).
+     */
+    @Override
+    public Object getFieldValue(Field field, Object originalValue)
+            throws RuntimeException
+    {
+        Types.MinorType fieldType = Types.getMinorTypeForArrowType(field.getType());
+        String fieldName = field.getName();
+        Object fieldValue;
+
+        if (originalValue instanceof HashMap && ((Map) originalValue).containsKey(fieldName)) {
+            fieldValue = ((Map) originalValue).get(fieldName);
+        }
+        else {
+            throw new RuntimeException("Field not found in Document: " + fieldName);
+        }
+
+        switch (fieldType) {
+            case LIST:
+                return coerceListField(field, fieldValue);
+            case STRUCT:
+                if (fieldValue instanceof HashMap) {
+                    // Both fieldType and fieldValue are nested structures => return as map.
+                    return fieldValue;
+                }
+                break;
+            default:
+                if (!(fieldValue instanceof HashMap)) {
+                    return coerceField(field, fieldValue);
+                }
+                break;
+        }
+
+        throw new RuntimeException("Invalid field value encountered in Document for field: " + field.toString() +
+                ",value: " + fieldValue.toString());
+    }
+
+    /**
+     * Allows for coercion of a list of values where the returned types do not match the schema.
+     * Multiple fields in Elasticsearch can be returned as a string, numeric (Integer, Long, Double), or null.
+     * @param field is the field that we are coercing the value into.
+     * @param fieldValue is the list of value to coerce
+     * @return the coerced list of value.
+     * @throws RuntimeException if the fieldType is not a LIST or the fieldValue is instanceof HashMap (STRUCT).
+     */
+    protected Object coerceListField(Field field, Object fieldValue)
+            throws RuntimeException
+    {
+        Types.MinorType fieldType = Types.getMinorTypeForArrowType(field.getType());
+
+        switch (fieldType) {
+            case LIST:
+                Field childField = field.getChildren().get(0);
+                if (fieldValue instanceof ArrayList) {
```

Comment:
`instanceof List` (in case underlying Arrow implementation changes)

Reply:
Fixed